rat/grammar.ebnf

```diff
@@ -37,10 +37,10 @@ factor
     | literal
 ;
 
-ifelse::IfElse = 'ifelse' ~ '(' predicate:expression ',' left:expression ',' right:expression ')';
+ifelse::IfElse = 'ifelse' '(' ~ predicate:expression ',' left:expression ',' right:expression ')';
 function_call::FunctionCall = name:identifier '(' [ arglist:arglist ] ')';
 variable::Variable = name:identifier [ constraints:constraints ] [ '[' arglist:arglist ']' ] [ prime:"'" ];
-constraints::Constraints = '<' ~ left:constraint ~ [',' right:constraint] '>';
+constraints::Constraints = '<' left:constraint ~ [',' right:constraint] '>';
```

Comment:
This was a bug that made it so that when the parser found:

`identifier <`, it thought that it had to match a constraint.

In reality that could also be `identifier < identifier`, or something else.

Reply:
Good catch!